app/src/main/java/io/homeassistant/companion/android/sensors/ChargingBroadcastReceiver.kt

```diff
@@ -0,0 +1,38 @@
+package io.homeassistant.companion.android.sensors
+
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.util.Log
+import io.homeassistant.companion.android.domain.integration.IntegrationUseCase
+import java.lang.Exception
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+
+class ChargingBroadcastReceiver(
+    private val integrationUseCase: IntegrationUseCase
+) : BroadcastReceiver() {
+    private val ioScope = CoroutineScope(Dispatchers.IO)
+    private var updateJob: Job? = null
+    override fun onReceive(context: Context, intent: Intent?) {
+        updateJob?.cancel()
+        updateJob = ioScope.launch {
+            val batterySensorManager = BatterySensorManager().registerSensors(context)
+            integrationUseCase.updateSensors(
+                batterySensorManager.getSensors(context).toTypedArray()
+            )
+        }
+    }
+
+    private suspend fun BatterySensorManager.registerSensors(context: Context) = apply {
+        getSensorRegistrations(context).forEach {
+            try {
```

Comment:
The other place where we do this (`SensorWorker`) also used this pattern. I had debated finding some way to avoid duplication of logic here but ended up not really being sure where to put that.